core/src/main/scala/org/apache/spark/partial/SumEvaluator.scala

```diff
@@ -42,6 +42,14 @@ private[spark] class SumEvaluator(totalOutputs: Int, confidence: Double)
       new BoundedDouble(counter.sum, 1.0, counter.sum, counter.sum)
     } else if (outputsMerged == 0) {
       new BoundedDouble(0, 0.0, Double.NegativeInfinity, Double.PositiveInfinity)
+    } else if (counter.count == 0) {
+      new BoundedDouble(0, 0.0, Double.NegativeInfinity, Double.PositiveInfinity)
+    } else if (counter.count == 1) {
+      val p = outputsMerged.toDouble / totalOutputs
```

Comment:
This duplicates code in the following block. I think you can compute the common values in one shared block and then branch afterwards to handle count == 1.